docs/architecture/blazor-for-web-forms-developers/security-authentication-authorization.md

```diff
@@ -1,15 +1,379 @@
+
+### Roles, claims, and policies
+
+Both the universal providers and ASP.NET Core Identity support the concept of roles. You can create roles for users and assign users to roles. Users can belong to any number of roles, and you can verify role membership as part of your authorization implementation.
+
+In addition to roles, ASP.NET Core identity supports the concepts of claims and policies. While a role should specifically correspond to a set of resources a user in that role should be able to access, a claim is simply part of a user's identity. A claim is a name value pair that represents what the subject is, not what the subject can do.
+
+It is possible to directly inspect a user's claims and determine based on these whether a user should be given access to a resource. However, such checks are often repetitive and scattered throughout the system. A better approach is to define a *policy*.
+
+An authorization policy consists of one or more requirements. Policies are registered as part of the authorization service configuration in the `ConfigureServices` method of `Startup.cs`. For example, the following code snippet configures a policy called "CanadiansOnly" which has the requirement that the user have the Country claim with the value of "Canada".
+
+```csharp
+services.AddAuthorization(options =>
+{
+    options.AddPolicy("CanadiansOnly", policy => policy.RequireClaim(ClaimTypes.Country, "Canada"));
+});
+```
+
+You can [learn more about how to create custom policies in the documentation](https://docs.microsoft.com/aspnet/core/security/authorization/policies).
+
+Whether you're using policies or roles, you can specify that a particular page in your Blazor application require that role or policy with the `[Authorize]` attribute, applied with the `@attribute` directive.
+
+Requiring a role:
+
+```csharp
+@attribute [Authorize(Roles ="administrators")]
+```
+
+Requiring a policy be satisfied:
+
+```csharp
+@attribute [Authorize(Policy ="CanadiansOnly")]
+```
+
+If you need access to a user's authentication state, roles, or claims you can access these by injecting the `AuthenticationStateProvider` into your component:
```

Comment:
This is correct, but often there's an easier way. In the Blazor templates we typically setup the [authentication state as a cascading parameter](https://docs.microsoft.com/aspnet/core/blazor/security#expose-the-authentication-state-as-a-cascading-parameter) that you can then bind to as a parameter in your component and the use in procedural logic.

Reply:
Looks like you mention this later in the migration section. I think it would be better to cover all of the core Blazor security concepts up front and then describe how to use them when migrating.